spec/classes/init_spec.rb

```diff
@@ -12,135 +12,148 @@
+          end
+
+          context 'modifying the banner' do
+            context 'disable' do
+              let(:params){{ :banner => false }}
 
+              it { is_expected.to compile.with_all_deps }
               it { is_expected.to create_dconf__settings('GDM Dconf Settings') }
               it {
                 dconf_resource = catalogue.resource('Dconf::Settings[GDM Dconf Settings]')
-                expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-text']['value']).to match(/ATTENTION/)
-                expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-enable']['value']).to be true
-              }
-              it {
-                is_expected.to create_gdm__set('GDM [chooser] Multicast:false').with(
-                  {
-                    :section => 'chooser',
-                    :key     => 'Multicast',
-                    :value   => 'false'
-                  }
-                )
+                expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-enable']['value']).to be false
               }
-              it { is_expected.to create_gdm__set('GDM [daemon] TimedLoginEnable:false') }
-              it { is_expected.to create_gdm__set('GDM [daemon] AutomaticLoginEnable:false') }
-              it { is_expected.to create_gdm__set('GDM [greeter] IncludeAll:false') }
-              it { is_expected.to create_gdm__set('GDM [security] DisallowTCP:true') }
-              it { is_expected.to create_gdm__set('GDM [xdmcp] Enable:false') }
             end
 
-            context 'modifying the banner' do
-              context 'disable' do
-                let(:params){{ :banner => false }}
+            context 'set new content' do
+              let(:params){{ :banner_content => 'I like banners' }}
+              let(:to_match){ "'I like banners'" }
+              let(:dconf_resource){ catalogue.resource('Dconf::Settings[GDM Dconf Settings]')}
 
-                it { is_expected.to compile.with_all_deps }
-                it { is_expected.to create_dconf__settings('GDM Dconf Settings') }
-                it {
-                  dconf_resource = catalogue.resource('Dconf::Settings[GDM Dconf Settings]')
-                  expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-enable']['value']).to be false
-                }
-              end
+              it { is_expected.to compile.with_all_deps }
+              it { is_expected.to create_dconf__settings('GDM Dconf Settings') }
+              it {
+                expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-enable']['value']).to be true
+                expect(dconf_resource[:settings_hash]['org/gnome/login-screen']['banner-message-text']['value']).to eq(to_match)
+              }
 
-              context 'set new content' do
+              context 'with pre-added quotes' do
```

Comment:
This test is the same as the previous context.  Why is it replicated here?

Reply:
I think what he was trying to test was a banner not bound by single quotes.  He wanted to make sure the code would add the single quotes.  I changed the second test from content='my banner'  to content ="my banner"  and left the test to check if it added the quotes.